R/ergm_mplecov.R

```diff
@@ -0,0 +1,146 @@
+#'   Clist.miss,
+#' @param verbose whether this and the C routines should be verbose (T
+#'   or F); default=FALSE
+#' 
+#' @param constraints {A formula specifying one or more constraints
+#' on the support of the distribution of the networks being modeled,
+#' using syntax similar to the \code{formula} argument, on the
+#' right-hand side.
+#' 
+#' @param family the family to use in the R native routine
+#' default="binomial"
+#' 
+#' @return \code{ergm_mplecov} returns a list either
+#'   containing a Godambe covariance matrix or a diagonal matrix with bootstrap variances.
+#' 
+#' @references Schmid CS and Desmarais BA (2017). "Exponential random graph 
+#' models with big networks: Maximum pseudolikelihood estimation and the parametric bootstrap" 
+#' _IEEE International Conference on Big Data (Big Data)_, pp. 116-121.
+#' 
+#' Schmid CS and Hunter DR (2023).  "Computing Pseudolikelihood Estimators for Exponential-Family Random Graph Models" _Journal of Data Science_.
+#' @noRd
+#' 
+#' @examples
+#' \donttest{
+#' # create initial network with x=50 nodes
+#' init.net <- network(x=50, directed = FALSE, density = 0.1)
+#' 
+#' # true parameters for edges, kstar(2), ad triangles
+#' truth <- c(-0.25, -0.2, 0.5)
+#' 
+#' # simulate a network from the ERGM defined by the true parameter values
+#' sim.net <- simulate(init.net~ edges+kstar(2)+ triangles, nsim=1, coef=truth)
+#' 
+#' # get MPLE with inverse Hessian matrix
+#' fisher <- ergm(sim.net ~ edges+kstar(2)+ triangles, estimate = "MPLE", 
+#' control = control.ergm(MPLE.covariance.method = "invHess"))
+#' 
+#' # get MPLE with Godambe matrix
+#' godambe <- ergm(sim.net ~ edges+kstar(2)+ triangles, estimate = "MPLE",
+#' control = control.ergm(MPLE.covariance.method = "Godambe"))
+#' 
+#' # get MPLE with bootstrap standard errors
+#' bootstrap <- ergm(sim.net ~ edges+kstar(2)+ triangles, estimate="MPLE",
+#' control = control.ergm(MPLE.covariance.method="bootstrap"))
+#' }
+ergm_mplecov <- function(pl,
+                         nw, 
+                         s, 
+                         init=init, 
+                         theta.mple, 
+                         invHess,  
+                         control=NULL,
+                         verbose=FALSE, 
+                         constraints=NULL, 
+                         family="binomial",
+                         formula = formula){
+  
+  m <- s$model
+  # get sample size from control.ergm
+  R <- control$MPLE.covariance.samplesize
+  mple.burnin <- control$MPLE.covariance.sim.burnin
+  mple.interval <- control$MPLE.covariance.sim.interval
+  
+  # Simulate R networks
+  sim.mple <- simulate(m, basis=nw, coef=theta.mple, nsim=R,
+                       control=control.simulate.formula(MCMC.burnin=mple.burnin, MCMC.interval=mple.interval))
+  
+  num.variables <- ncol(pl$xmat)
+  
+  terms.form <- terms(formula)
+  
+  new.formula <- reformulate(attr(terms.form, "term.labels"), "sim.mple[[i]]")
```

Comment:
This could affect the formula in unexpected ways. Suggest leaving the LHS alone and using the `basis` argument where needed.